### PR TITLE
feat: standardize plugins

### PR DESCRIPTION
**Pull request by AllanFly120**
Standardizes naming of plugins. continue to #402 

/cc @srchase 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: clients/client-rds-data/README.md [hunk 1]
```diff
@@ -18,29 +18,29 @@ Serverless</a> in the <i>Amazon Aurora User Guide</i>.</p>
...
 //JavaScript
 const {
-  RDSDataClient
-} = require("@aws-sdk/client-rds-data-node/RDSDataClient");
+  RdsDataServiceClient
+} = require("@aws-sdk/client-rds-data/RdsDataServiceClient");
 const {
   BatchExecuteStatementCommand
-} = require("@aws-sdk/client-rds-data-node/commands/BatchExecuteStatementCommand");
+} = require("@aws-sdk/client-rds-data/commands/BatchExecuteStatementCommand");
```
Comment (trivikr) [inline clients/client-rds-data/README.md]: nit: import from root
```javascript
const {
  RDSDataServiceClient,
  BatchExecuteStatementCommand
} = require("@aws-sdk/client-rds-data");
```

File: clients/client-rds-data/README.md [hunk 2]
```diff
@@ -18,29 +18,29 @@ Serverless</a> in the <i>Amazon Aurora User Guide</i>.</p>
...
 ```
 
 ```javascript
 //TypeScript
-import { RDSDataClient } from "@aws-sdk/client-rds-data-node/RDSDataClient";
-import { BatchExecuteStatementCommand } from "@aws-sdk/client-rds-data-node/commands/BatchExecuteStatementCommand";
+import { RdsDataServiceClient } from "@aws-sdk/client-rds-data/RdsDataServiceClient";
+import { BatchExecuteStatementCommand } from "@aws-sdk/client-rds-data/commands/BatchExecuteStatementCommand";
```
Comment (trivikr) [inline clients/client-rds-data/README.md]: nit: import from root:
```typescript
import {
  RdsDataServiceClient,
  BatchExecuteStatementCommand
} from "@aws-sdk/client-rds-data";
```

File: clients/client-rds-data/README.md [hunk 3]
```diff
@@ -90,7 +90,7 @@ rDSData.send(batchExecuteStatementCommand, (err, data) => {
 The SDK can also send requests using the simplified callback style from version 2 of the SDK.
```
Comment (trivikr) [inline clients/client-rds-data/README.md]: nit: remove this section, as we don't plan to provide simplified callback style?

File: clients/client-rds-data/RdsDataServiceClient.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+      ...RDSRuntimeConfiguration,
+      ...configuration
+    });
+    let _config_1 = resolveRegionConfig(_config_0);
+    let _config_2 = resolveAwsAuthConfig(_config_1);
+    let _config_3 = resolveEndpointsConfig(_config_2);
+    let _config_4 = resolveRetryConfig(_config_3);
+    let _config_5 = resolveUserAgentConfig(_config_4);
+    super(_config_5);
+    this.config = _config_5;
```
Comment (trivikr) [inline clients/client-rds-data/RdsDataServiceClient.ts]: nit: The below suggestion avoids creation of temporary variables. Can we use it instead if it's not difficult to CodeGen?
```javascript
this.config = resolveUserAgentConfig(
  resolveRetryConfig(
    resolveEndpointsConfig(
      resolveAwsAuthConfig(
        resolveRegionConfig(
          resolveClientProtocolConfig({
            ...RDSRuntimeConfiguration,
            ...configuration
          })
        )
      )
    )
  )
);
super(this.config);

```

Comment (trivikr) [reply clients/client-rds-data/RdsDataServiceClient.ts]: A simpler way would be to use generic compose function for this ([docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#higher-order-type-inference-from-generic-functions))

Needs testing:
```typescript
function compose<A, B, C>(f: (arg: A) => B, g: (arg: B) => C): (arg: A) => C {
    return x => g(f(x));
}

this.config = compose(
  resolveUserAgentConfig,
  resolveRetryConfig,
  resolveEndpointsConfig,
  resolveAwsAuthConfig,
  resolveRegionConfig,
  resolveClientProtocolConfig
)({
  ...RDSRuntimeConfiguration,
  ...configuration
});

```

Comment (AllanFly120) [reply clients/client-rds-data/RdsDataServiceClient.ts]: I think this compose interface needs [Variadic Kinds](https://github.com/Microsoft/TypeScript/issues/5453) to yet implemented in TS. Currently there's no way to spreading the generic in TS IMO. For example:

```typescript
function compose<...T>//and other things
```

**Pull request merged**